impeller/renderer/backend/metal/BUILD.gn

```diff
@@ -5,6 +5,9 @@
 import("../../../tools/impeller.gni")
 
 impeller_component("metal") {
+  cflags_objc = flutter_cflags_objc_arc
+  cflags_objcc = flutter_cflags_objcc_arc
```

Comment:
I skimmed some of the files in the list below, and a bunch have Obj-C ivars ([example](https://github.com/flutter/engine/blob/382fd1ea143afd57c1191500d75ac23984e939cf/impeller/renderer/backend/metal/blit_pass_mtl.h#L22-L24)). Is implicitly changing those from weak to strong intentional?

Reply:
Re-auditing, you're right there are some in here that should be weak. There were a lot of cases where they should be strong -- i.e. pre-ARC code was allocating but never releasing objects, which I had been using as a shortcut to find the strong objects; looks like that shortcut doesn't apply well here. I'm going to break this patch up.

Reply:
Reverted -- only change now is standardisation of flags and ARC-ificiation of simple unit tests that hadn't been migrated. Impeller to follow.